docs: Generate SQL diagrams for PAUSE/RESUME/CANCEL JOB and EXPLAIN/EXPLAIN ANALYZE

lnhsingh opened the pull request:
Release note: None

Commit:
Generate SQL diagrams for PAUSE/RESUME/CANCEL JOB

edit diagrams

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/30036)
<!-- Reviewable:end -->

lhirata commented:
bors r+

lhirata commented:
@mjibson can i merge this once bors runs? or do i need to do something special?

lhirata commented:
bors r+

mjibson commented:
bors r+

Commit:
Add EXPLAIN ANALYZE diagram

Release notes: None

Rebase with master / regenerate diagram

Minor edit to remove double line

mjibson commented:
You need to rebase on master and rebuild the BNF files because some things changed. I can do or help out with this if you need because it may be tricky. The other option is to just make a brand new branch from master and apply the changes to diagrams.go and submit a new PR.

lhirata commented:
Thanks @mjibson. I think I was able to rebase correctly, but lmk if something looks off.

lhirata commented:
@mjibson - when you have time, can you review the additional diagram changes?

mjibson commented on pkg/cmd/docgen/diagrams.go:
```diff
@@ -677,8 +677,20 @@ var specs = []stmtSpec{
 		inline: []string{"explain_option_list"},
 		replace: map[string]string{
 			"explain_option_name": "( 'VERBOSE' | 'TYPES' | 'OPT' | 'DISTSQL' )",
+			"'ANALYZE'":           "",
```
Instead of a replace rule, you should use an exclude rule here which will prevent the double identical lines:

`exclude: []*regexp.Regexp{regexp.MustCompile("'ANALYZE'")},`

lnhsingh reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/30036)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/docgen/diagrams.go, line 680 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/30036#-LM4P0LwieQWCPQ2SUtY-r2-680:-LONwW0o0WwbIaVCDjG5:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8264df6d0eca2ee65a52d741a6225d4fe29ff361/pkg/cmd/docgen/diagrams.go#L680)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

Instead of a replace rule, you should use an exclude rule here which will prevent the double identical lines:

`exclude: []*regexp.Regexp{regexp.MustCompile("'ANALYZE'")},`
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

lhirata commented:
bors r+
